app/storyedit/models/story.model.ts

```diff
@@ -69,57 +125,59 @@ export class StoryModel {
     return this.doc.destroy().map(() => { return true; });
   }
 
-  get isValid(): boolean {
-    for (let key in this.invalid) {
-      if (this.invalid[key]) { return false; }
-    }
-    return true;
-  }
-
-  get isChanged(): boolean {
-    for (let key in this.changed) {
-      if (this.changed[key]) { return true; }
+  invalid(fieldName?: string): string {
+    if (fieldName === 'title') {
+      if (!this.title || this.title.length < 1) {
+        return 'is required';
+      } else if (this.title.length < 10) {
+        return 'is too short';
+      }
+    } else if (fieldName === 'shortDescription') {
+      if (!this.shortDescription || this.shortDescription.length < 1) {
+        return 'is required';
+      } else if (this.shortDescription.length < 10) {
+        return 'is too short';
+      }
+    } else if (fieldName === 'genre') {
+      if (!this.genre || this.genre.length < 1) {
+        return 'is required';
+      }
+    } else if (fieldName === 'subGenre') {
+      if (!this.subGenre || this.subGenre.length < 1) {
+        return 'is required';
+      }
+    } else if (fieldName === 'extraTags') {
+      let tags = (this.extraTags || '').replace(/\s*,\s*/, ',').split(',');
+      for (let tag of tags) {
+        if (tag && tag.length < 3) {
+          return `must contain at least 3 characters per tag (see ${tag})`;
+        }
+      }
+    } else if (fieldName === 'tags') {
+      if (this.invalid('genre')) {
+        return `Genre ${this.invalid('genre')}`;
+      } else if (this.invalid('subGenre')) {
+        return `SubGenre ${this.invalid('subGenre')}`;
+      } else if (this.invalid('extraTags')) {
+        return `Tags ${this.invalid('extraTags')}`;
+      }
+    } else if (!fieldName) {
+      return this.invalid('title') || this.invalid('shortDescription') ||
+             this.invalid('genre') || this.invalid('subGenre');
+    } else {
+      return null;
     }
```

Comment:
This seems like a pretty different approach from the ng2 control and control groups stuff, like this or something -  http://blog.thoughtram.io/angular/2016/03/14/custom-validators-in-angular-2.html

Did you try this style of form builder and it was lacking?